pkg/didcomm/protocol/didexchange/states_test.go

```diff
@@ -1008,6 +897,87 @@ func createDIDDocWithKey(pub string) *diddoc.Doc {
 	return didDoc
 }
 
+func getProvider() (protocol.MockProvider, *mockstorage.MockStore) {
+	return protocol.MockProvider{}, &mockstorage.MockStore{Store: make(map[string][]byte)}
+}
+
+func getContext(prov protocol.MockProvider, store storage.Store) *context {
+	pubKey, privKey := generateKeyPair()
+
+	return &context{outboundDispatcher: prov.OutboundDispatcher(),
+		didCreator:      &mockdid.MockDIDCreator{Doc: createDIDDocWithKey(pubKey)},
+		signer:          &mockSigner{privateKey: privKey},
+		connectionStore: NewConnectionRecorder(store, store),
+		didStore:        prov.DIDStore(),
+		didResolver:     prov.DIDResolver(),
+	}
+}
+
+func createRequest(ctx *context) (*Request, error) {
+	newDidDoc, err := ctx.didCreator.Create(testMethod)
+	if err != nil {
+		return nil, err
+	}
+	// Prepare did-exchange inbound request
+	request := &Request{
+		Type:  RequestMsgType,
+		ID:    randomString(),
+		Label: "Bob",
+		Connection: &Connection{
+			DID:    newDidDoc.ID,
+			DIDDoc: newDidDoc,
+		},
+	}
+
+	return request, nil
+}
+
+func createResponse(newDidDoc *diddoc.Doc, id string, ctx *context) (*Response, error) {
+	connection := &Connection{
+		DID:    newDidDoc.ID,
+		DIDDoc: newDidDoc,
+	}
+	connectionSignature, err := ctx.prepareConnectionSignature(connection)
+
+	if err != nil {
+		return nil, err
+	}
+
+	response := &Response{
+		Type: ResponseMsgType,
+		ID:   randomString(),
+		Thread: &decorator.Thread{
+			ID: id,
+		},
+		ConnectionSignature: connectionSignature,
+	}
+
+	return response, nil
+}
+
+func saveConnectionRecord(newDidDoc *diddoc.Doc, ctx *context) (*Response, error) {
+	response, err := createResponse(newDidDoc, generateRandomID(), ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	connRec := &ConnectionRecord{State: (&responded{}).Name(), ThreadID: response.Thread.ID,
+		ConnectionID: "123"}
```

Comment:
nit: next time please split long structure like this (one field in one line)
```
connRec := &ConnectionRecord{
    State: (&responded{}).Name(), 
    ThreadID: response.Thread.ID,
    ConnectionID: "123",
}
```

Reply:
done